samples/nrf9160/at_cmd_parser/src/main.c

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr.h>
+
+#include <at_cmd_interface.h>
+#include <at_cmd_decoder.h>
+
+/* This callback will be called when ever a message from the modem is completely
+ * parsed. The model parameter will contain an identifier for the content
+ * of the message, and the model_ptr parameter will contain a pointer to a
+ * model of the data.
+ */
+void process_model(enum at_cmd_models model, void *model_ptr)
+{
+	static u32_t count;
+
+	printk("------Message %04d-------\n", count++);
+
+	switch (model) {
+	case AT_CMD_CEREG_MODEL:
+		printk("CEREG Stat: %d\n",
+			((struct at_cmd_model_cereg *)model_ptr)->stat);
```

Comment:
Please use a local variable to make the code readable.

Reply:
I am not sure which is the more readable? 

With your suggestion I would need to create a pointer variable for each model I want to use in the top of the function, and I would still need to cast the pointer at least once

With the current implementation it is easy to see which model is being used, and I don't need to allocate a pointer variable for each possible model.